src/main/java/org/joda/time/field/FieldUtils.java

```diff
@@ -177,6 +177,22 @@ public static long safeMultiply(long val1, long val2) {
     }
     
     /**
+     * Divides the dividend by the divisor throwing an exception if 
+     * overflow occurs or the divisor is zero.
+     * 
+     * @param dividend  the dividend
+     * @param divisor  the divisor
+     * @return the new total
+     * @throws ArithmeticException if the operation overflows or the divisor is zero
+     */
+    public static long safeDivide(long dividend, long divisor)
+    {
+        if(dividend == Long.MIN_VALUE && divisor == -1L)
```

Comment:
Why is the check divisor == -1L given here? Should it not be divisor == 0L ? Kindly help me understand this